### PR TITLE
Refactor message transmission logic

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Move sending logic from controller into its own serivce
* Create service contract/interface to decouple HTTP controller from service implementation
* Decompose logic into multiple small, readable methods
* Wrap data in immutable data classes/objects to prevent long parameter lists and easy testing
* Add basic unit tests that cover the main code paths

TODO:
- [x] manual testing of various use cases
  - [x] send new message
  - [x] reply to a message
  - [x] attach files on a new message
  - [x] attach files on an existing messageb
- [x] AccountController tests

### CODE REVIEW

File: lib/Contracts/IMailTransmission.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace OCA\Mail\Contracts;
+
+use OCA\Mail\Db\Alias;
+use OCA\Mail\Model\NewMessageData;
+use OCA\Mail\Model\RepliedMessageData;
+
+interface IMailTransmission {
+
+	public function sendMessage(NewMessageData $message, RepliedMessageData $reply, Alias $alias = null, $draftUID = null);
```
Comment (ChristophWurst) [inline lib/Contracts/IMailTransmission.php]: - [ ] TODO: comments

**Comment by ChristophWurst** [comment]
bildschirmfoto von 2017-07-10 13-47-43

🤓 🚀 🤖

**Comment by ChristophWurst** [comment]
bildschirmfoto von 2017-07-10 15-20-51

😀

**Comment by ChristophWurst** [comment]
@nextcloud/mail commits squashed and ready to be reviewed!

**Comment by cmuench** [comment]
👍

**Comment by jancborchardt** [comment]
@cmuench any testing & review of other pull requests is also much appreciated! :)

**Pull request merged**